Apps/WebClient/src/ClientApp/src/views/pcrTest.vue

```diff
@@ -573,44 +582,45 @@ export default class PCRTest extends Vue {
             </b-row>
             <b-row>
                 <b-col>
-                    <form>
+                    <form @submit.prevent="handleSubmit">
                         <!-- NAME -->
                         <b-row v-if="pcrDataSource === DSKEYCLOAK" class="pt-2">
                             <b-col>
                                 <label for="pcrTestFullName">Name:</label>
                                 <strong id="prcTestFullName">
-                                    {{ fullName }}
+                                    {{ getfullName() }}
                                 </strong>
                             </b-col>
                         </b-row>
-                        <b-row v-if="noSerialNumber" class="mt-2">
+                        <b-row v-if="noTestKitCode" class="mt-2">
                             <b-col>
-                                <label for="testKitId">
-                                    Test Kit Serial Number
-                                </label>
+                                <label for="testKitCode"> Test Kit Code </label>
                                 <b-form-input
-                                    id="testKitId"
-                                    v-model="pcrTest.testKitId"
+                                    id="testKitCode"
+                                    v-model="pcrTest.testKitCode"
+                                    data-testid="test-kit-code-input"
                                     type="text"
-                                    placeholder="Serial Number"
-                                    :state="isValid($v.pcrTest.testKitId)"
-                                    @blur.native="$v.pcrTest.testKitId.$touch()"
+                                    placeholder="Test Kit Code"
+                                    :state="isValid($v.pcrTest.testKitCode)"
+                                    @blur.native="
+                                        $v.pcrTest.testKitCode.$touch()
+                                    "
                                 />
                                 <b-form-invalid-feedback
                                     v-if="
-                                        $v.pcrTest.testKitId.$dirty &&
-                                        !$v.pcrTest.testKitId.required
+                                        $v.pcrTest.testKitCode.$dirty &&
+                                        !$v.pcrTest.testKitCode.formatted
```

Comment:
It looks like we display an error when the test kit fails the formatted validator but not when it fails the required validator.